Connected System Plug-in (CSP) Examples/Google Drive (OAuth,  Sending Files and Multiple Operations)/src/main/java/com/mycorp/googledrive/templates/GoogleDriveUploadFileIntegrationTemplate.java

```diff
@@ -0,0 +1,123 @@
+import com.appian.connectedsystems.templateframework.sdk.TemplateId;
+import com.appian.connectedsystems.templateframework.sdk.configuration.Document;
+import com.appian.connectedsystems.templateframework.sdk.configuration.FolderPropertyDescriptor;
+import com.appian.connectedsystems.templateframework.sdk.configuration.PropertyPath;
+import com.appian.connectedsystems.templateframework.sdk.diagnostics.IntegrationDesignerDiagnostic;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.drive.Drive;
+import com.google.api.services.drive.model.File;
+import com.google.common.base.Stopwatch;
+
+@TemplateId(name = "GoogleDriveUploadFileIntegrationTemplate")
+public class GoogleDriveUploadFileIntegrationTemplate extends SimpleIntegrationTemplate {
+
+  public static final String FILE_ID_KEY = "fileId";
+  public static final String FOLDER_ID_KEY = "folderID";
+
+  @Override
+  protected SimpleConfiguration getConfiguration(
+      SimpleConfiguration integrationConfiguration,
+      SimpleConfiguration connectedSystemConfiguration,
+      PropertyPath updatedProperty,
+      ExecutionContext executionContext) {
+    return integrationConfiguration.setProperties(
+        textProperty(FILE_ID_KEY)
+            .label("File ID")
+            .instructionText("This can be found in a shareable link for a file as part of the URL, e.g. " +
+                "https://drive.google.com/open?id=<FileID>")
+            .build(),
+        FolderPropertyDescriptor.builder()
+            .key(FOLDER_ID_KEY)
+            .label("Save to Folder")
+            .build()
+    );
+  }
+
+  @Override
+  protected IntegrationResponse execute(
+      SimpleConfiguration integrationConfiguration,
+      SimpleConfiguration connectedSystemConfiguration,
+      ExecutionContext executionContext) {
+    String fileId = integrationConfiguration.getValue(FILE_ID_KEY);
+    Long folderId = integrationConfiguration.getValue(FOLDER_ID_KEY);
+
+    IntegrationDesignerDiagnostic.IntegrationDesignerDiagnosticBuilder diagnosticBuilder = IntegrationDesignerDiagnostic
+        .builder();
+    Stopwatch stopwatch = Stopwatch.createStarted();
+
+    //GoogleCredential is used to create a Google client with Drive. To send a file, you need to supply a File
+    //and an InputStreamContent.
+    GoogleCredential credential = new GoogleCredential().setAccessToken(
+        executionContext.getAccessToken().get());
+    Drive drive = new Drive.Builder(new NetHttpTransport(), JacksonFactory.getDefaultInstance(),
+        credential).build();
+
+    File fileMetadata;
+    OutputStream outputStream = new ByteArrayOutputStream();
+    try {
+      fileMetadata = drive.files().get(fileId).execute();
+      drive.files().get(fileId).executeMediaAndDownloadTo(outputStream);
```

Comment:
I think there is a method called "executeMediaToInputStream" or something like that. Use that instead.